packages/openapi-ts/test/__snapshots__/test/generated/v3_angular_transform/types.gen.ts.snap

```diff
@@ -0,0 +1,132 @@
+// This file is auto-generated by @hey-api/openapi-ts
+
+/**
+ * This is a model that contains a some dates
+ */
+export type SimpleModel = {
+    id: number;
+    name: string;
+    readonly enabled: boolean;
+};
+
+/**
+ * This is a model that contains a some dates
+ */
+export type ModelWithDates = {
+    id: number;
+    name: string;
+    readonly enabled: boolean;
+    readonly modified: Date;
+    readonly expires?: Date;
+};
+
+export function ModelWithDates(data: any): ModelWithDates {
+    if (data?.modified) {
+        data.modified = new Date(data.modified);
+    }
+    if (data?.expires) {
+        data.expires = new Date(data.expires);
+    }
+    return data;
+}
+
+/**
+ * This is a model that contains a some dates and arrays
+ */
+export type ParentModelWithDates = {
+    id: number;
+    readonly modified?: Date;
+    items?: Array<ModelWithDates>;
+    item?: ModelWithDates;
+    simpleItems?: Array<SimpleModel>;
+    simpleItem?: SimpleModel;
+    dates?: Array<(Date)>;
+    strings?: Array<(string)>;
+};
+
+export function ParentModelWithDates(data: any): ParentModelWithDates {
+    if (data?.modified) {
+        data.modified = new Date(data.modified);
+    }
+    if (Array.isArray(data?.items)) {
+        data.items.forEach(ModelWithDates);
```

Comment:
Wouldn't you want to map this array too and return mapped items?

Reply:
All of the transformers actually mutate the object references in place, so having a return in general isn't really needed and a map is semantically incorrect

This is fantastic for runtime performance because we're not generating a bunch of new arrays and objects, and no other code has a handle to the object before this function has completed so immutability isn't really a concern